src/tasks/gitea/gitea-drone-oauth.ts

```diff
@@ -0,0 +1,212 @@
+import * as k8s from '@kubernetes/client-node'
+import axios, { AxiosRequestConfig } from 'axios'
+import querystring from 'querystring'
+
+import { UserApi, CreateOAuth2ApplicationOptions, OAuth2Application } from '@redkubes/gitea-client-node'
+
+import { cleanEnv, GITEA_USER, GITEA_PASSWORD, GITEA_URL, GITEA_REPO } from '../../validators'
+
+const env = cleanEnv({
+  GITEA_USER,
+  GITEA_PASSWORD,
+  GITEA_URL,
+  GITEA_REPO,
+})
+type DroneSecret = {
+  clientId: string
+  clientSecret: string
+}
+class GiteaDroneOAuth {
```

Comment:
We promote functional programming over OOP.
But it always good to discuss why a given approach has been taken. Could you elaborate a bit?

Reply:
I see no point in this setup to use a class. Please use functions.

Reply:
Reason I set it up as a class is because I took inspiration from another script (in otomi-api)
https://github.com/redkubes/otomi-api/blob/master/src/otomi-stack.ts#L196-L202

To get the api client, and reuse it, instead of creating it over-and-over-again.

Reply:
It makes sense to use objects if for example you apply dependency injection pattern or you just want to obtain object that encapsulate functionality, because it is used in some complex application. But here, in this very simple script, I recommend to use functions.